vector/src/main/java/im/vector/app/core/error/ErrorFormatter.kt

```diff
@@ -116,6 +116,10 @@ class DefaultErrorFormatter @Inject constructor(
                     throwable.error.code == MatrixError.M_PASSWORD_NO_SYMBOL -> {
                         stringProvider.getString(R.string.tchap_register_pwd_no_symbol)
                     }
+                    throwable.error.code == MatrixError.M_UNKNOWN &&
+                            throwable.error.message == "Not allowed to join this room" -> {
+                        stringProvider.getString(R.string.tchap_room_access_unauthorized_to_externs)
```

Comment:
@ClaireGizard
- the error is not limited to extern account, so please remove "to_externs" string from the key
- I realized we should apply this change at the Element level to let the community translate this message
(like [above](https://github.com/tchapgouv/tchap-android-v2/pull/344/files#diff-30b3061a248298e904cf64768291a06f9da2913617e717489c377bc6a01dcb26R99)
For that, prepare a PR in Element-Android by introducing a new string key "room_error_access_unauthorized". I suggest add it [here](https://github.com/vector-im/element-android/blob/f4cfb5d6d3874df80df405505e5e9cdc9df5602a/vector/src/main/res/values/strings.xml#L171)